### ISSUE TITLE
Loss function modification

### ISSUE THREAD

**Issue by groot-1313** [aurora95/Keras-FCN#44]
I have 300x300 single channel images. I am trying a segmentation problem. I have two classes - object1 and object2, and the rest is background. Therefore, my output is of size 300x300x2. 1st channel has 0s representing background and object2, and 1s representing object1. 2nd channel has 0s representing background and object1, and 1s representing object2. What changes are required in the function in order for me to be able to use it, because your loss function involves finding the one hot vector from the output obtained, and I believe my output is already in that form.

**Comment by mrgloom** [comment]
I think in case of `softmax+categorical_crossentropy` you need 3 planes [object1,object2,background] in case of `sigmoid+binary_crossentropy`  2 planes [object1,object2].
